### PR TITLE
Add `background` arg in `draw` function

### LINKED ISSUE #13: Add `background` argument to `draw` function
Currently, the `draw` function draws the circuit in color black without drawing the background. This may cause problems if the user is working in an environment with a dark background color. 

As an example, here is a drawing inside VS Code using the color theme _Dark+ (default light)_:
image

A straightforward solution would be to create an optional argument to `draw` function in which we could specify the background color of the output.

### PR DESCRIPTION
**Pull request by jofrevalles**
Fix #13.

### Summary
Now the function `draw` accepts the `background` argument, which is a `::String` that defines the background color of the canvas.

### Example
```julia
julia> circ = Circuit(4); push!(circ, Sd(1)); push!(circ, CX(2,4)); push!(circ, CZ(1,4));
julia> draw(circ, background="turquoise")
```
Output:
image

### CODE REVIEW

File: src/Visualization.jl [hunk 1]
```diff
@@ -47,20 +47,21 @@ function Base.show(io::IO, ::MIME"image/svg+xml", circuit::Circuit)
...
+function draw(gate::Gate; top::Bool = false, bottom::Bool = false, background::Union{String, Nothing} = nothing)
     n = (length ∘ lanes)(gate)
     if n == 1
-        draw_block(; top = top, bottom = bottom)
+        draw_block(; top = top, bottom = bottom, background)
     else
         a, b = extrema(lanes(gate))
         n = b - a + 1
-        vcat(draw_block(; top = top), fill(draw_multiblock_mid(), (n - 2))..., draw_block(; bottom = bottom))
+        vcat(draw_block(; top = top, background), fill(draw_multiblock_mid(; background), (n - 2))..., draw_block(; bottom = bottom, background))
```
Comment (mofeing) [inline src/Visualization.jl:57]:
```suggestion
        vcat(draw_block(; top = top, kwargs...), fill(draw_multiblock_mid(; kwargs...), (n - 2))..., draw_block(; bottom = bottom, kwargs...))
```

File: src/Visualization.jl [hunk 2]
```diff
@@ -73,7 +74,7 @@ for (T, N) in [(X, "X"), (Y, "Y"), (Z, "Z"), (H, "H"), (S, "S"), (Sd, L"S^\dagge
     end
 end
 
-function draw(gate::Control)
+function draw(gate::Control; background::Union{String, Nothing} = nothing)
```
Comment (mofeing) [inline src/Visualization.jl:77]:
```suggestion
function draw(gate::Control; kwargs...)
```

File: src/Visualization.jl [hunk 3]
```diff
@@ -47,20 +47,21 @@ function Base.show(io::IO, ::MIME"image/svg+xml", circuit::Circuit)
...
         a, b = extrema(lanes(gate))
         n = b - a + 1
-        vcat(draw_block(; top = top), fill(draw_multiblock_mid(), (n - 2))..., draw_block(; bottom = bottom))
+        vcat(draw_block(; top = top, background), fill(draw_multiblock_mid(; background), (n - 2))..., draw_block(; bottom = bottom, background))
     end
 end
 
 draw(::I) = draw(I)
-function draw(::Type{I})
+function draw(::Type{I}; background::Union{String, Nothing} = nothing)
```
Comment (mofeing) [inline src/Visualization.jl:62]:
```suggestion
function draw(::Type{I}; background = nothing)
```

File: src/Visualization.jl [hunk 4]
```diff
@@ -83,28 +84,29 @@ function draw(gate::Control)
...
-                draw_cross()
+                draw_cross(; background)
             end for lane in r if lane < only(t)
         ]...,
-        draw(op(gate); top = !any(<(only(t)), c), bottom = !any(>(only(t)), c)),
+        draw(op(gate); top = !any(<(only(t)), c), bottom = !any(>(only(t)), c), background),
         [
             if lane == last(r)
-                draw_copy(:bottom)
+                draw_copy(:bottom; background)
```
Comment (mofeing) [inline src/Visualization.jl:97]:
```suggestion
                draw_copy(:bottom; kwargs...)
```

File: src/Visualization.jl [hunk 5]
```diff
@@ -83,28 +84,29 @@ function draw(gate::Control)
...
             elseif lane ∈ c
-                draw_copy(:mid)
+                draw_copy(:mid; background)
```
Comment (mofeing) [inline src/Visualization.jl:99]:
```suggestion
                draw_copy(:mid; kwargs...)
```

File: src/Visualization.jl [hunk 6]
```diff
@@ -83,28 +84,29 @@ function draw(gate::Control)
...
             else
-                draw_cross()
+                draw_cross(; background)
```
Comment (mofeing) [inline src/Visualization.jl:101]:
```suggestion
                draw_cross(; kwargs...)
```

File: src/Visualization.jl [hunk 7]
```diff
@@ -83,28 +84,29 @@ function draw(gate::Control)
...
             end for lane in r if lane > only(t)
         ]...,
     )
 end
 
-function draw_block(label = ""; top::Bool = false, bottom::Bool = false)
+function draw_block(label = ""; top::Bool = false, bottom::Bool = false, background::Union{String, Nothing} = nothing)
```
Comment (mofeing) [inline src/Visualization.jl:107]:
```suggestion
function draw_block(label = ""; top::Bool = false, bottom::Bool = false, background = nothing)
```

File: src/Visualization.jl [hunk 8]
```diff
@@ -127,8 +129,9 @@ function draw_block(label = ""; top::Bool = false, bottom::Bool = false)
     end 50 50
 end
 
-function draw_multiblock_mid()
+function draw_multiblock_mid(;background::Union{String, Nothing} = nothing)
```
Comment (mofeing) [inline src/Visualization.jl:132]:
```suggestion
function draw_multiblock_mid(; background = nothing)
```

File: src/Visualization.jl [hunk 9]
```diff
@@ -141,17 +144,19 @@ function draw_multiblock_mid()
     end 50 50
 end
 
-function draw_cross()
+function draw_cross(;background::Union{String, Nothing} = nothing)
```
Comment (mofeing) [inline src/Visualization.jl:147]:
```suggestion
function draw_cross(; background = nothing)
```

File: src/Visualization.jl [hunk 10]
```diff
@@ -141,17 +144,19 @@ function draw_multiblock_mid()
...
+        (background !== nothing) && Luxor.background(background)
         origin()
 
         line(Point(-25, 0), Point(25, 0), action = :stroke)
         line(Point(0, -25), Point(0, 25), action = :stroke)
     end 50 50
 end
 
-function draw_copy(dir::Symbol)
+function draw_copy(dir::Symbol; background::Union{String, Nothing} = nothing)
```
Comment (mofeing) [inline src/Visualization.jl:157]:
```suggestion
function draw_copy(dir::Symbol; background = nothing)
```

File: src/Visualization.jl [hunk 11]
```diff
@@ -4,11 +4,11 @@ using MathTeXEngine
 function draw end
 export draw
 
-function draw(circuit::Circuit)
+function draw(circuit::Circuit; background::Union{String, Nothing} = nothing)
     n = lanes(circuit)
 
     if isempty(circuit)
-        return vcat([draw(I) for _ in 1:n]...)
+        return vcat([draw(I; background) for _ in 1:n]...)
```
Comment (mofeing) [inline src/Visualization.jl:11]:
```suggestion
        return vcat([draw(I; kwargs...) for _ in 1:n]...)
```

File: src/Visualization.jl [hunk 12]
```diff
@@ -38,7 +38,7 @@ function draw(circuit::Circuit)
         (min, max) = extrema(mapreduce(lanes, ∪, moment))
         moment = [map(I, filter(<(min), 1:n))..., moment..., map(I, filter(>(max), 1:n))...]
 
-        mapreduce(draw, vcat, moment)
+        mapreduce(x->draw(x; background), vcat, moment)
```
Comment (mofeing) [inline src/Visualization.jl:41]:
```suggestion
        mapreduce(x->draw(x; kwargs...), vcat, moment)
```

**Review by mofeing — CHANGES_REQUESTED** [review]
Please revise these change before I merge them.

**Review by jofrevalles — COMMENTED** [review]
I agree with you, this a better way to fix the issue. I tested it and can confirm it works.

**Comment by mofeing** [comment]
@jofrevalles I think you marked my changes as resolved but didn't accept them

**Pull request merged**